### ISSUE TITLE
identity_conversion help string broke

### ISSUE THREAD

**Issue by CAD97** [rust-lang-nursery/rust-clippy#2971]
https://github.com/Hopson97/Asciimon/blob/0d0909011aebf87fedc2132e7869abd668702610/src/graphics/colour.rs#L26-L28

```rust
        String::from(format!(
            "\x1b[{};2;{};{};{}m", id, r, g, b
         ))
```

```
warning: identical conversion
  --> src\graphics\colour.rs:26:9
   |
26 | /         String::from(format!(
27 | |             "\x1b[{};2;{};{};{}m", id, r, g, b
28 | |         ))
   | |__________^ help: consider removing `String::from()`: `$ crate :: fmt :: format ( format_args ! ( $ ( $ arg ) * ) )`
   |
   = note: #[warn(identity_conversion)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#identity_conversion
```

**Comment by oli-obk** [comment]
We should probably add some way to get the outer most span when looking for suggestions. Macro expanded spans are never helpful
